### PR TITLE
feat: add cancel button

### PR DESCRIPTION
**Pull request by jon4hz**

### CODE REVIEW

File: src/app/sidebar/sidebar/sidebar.component.html [hunk 1]
```diff
@@ -21,6 +21,7 @@
               @switch (mapDownloadStates[layer.key]) {
                 @case ('loading') {
                   <mat-progress-bar color="primary" mode="indeterminate" />
```
Comment (jannis-z) [inline src/app/sidebar/sidebar/sidebar.component.html:23]: add spacer

File: src/app/sidebar/sidebar/sidebar.component.ts [hunk 2]
```diff
@@ -114,38 +116,62 @@ export class SidebarComponent {
...
+      url: downloadUrl,
+      map,
+      mapStatus: 'missing',
+      objectUrl: undefined,
+      mapStyle: undefined,
+    };
+
+    await db.localMapMeta.put(localMapMeta);
+
+    if (localMapMeta.mapStatus == 'missing') {
```
Comment (jannis-z) [inline src/app/sidebar/sidebar/sidebar.component.ts:129]: statement is not necessary. after click on download, it should always change to 'loading'.

File: src/app/sidebar/sidebar/sidebar.component.ts [hunk 3]
```diff
@@ -114,38 +116,62 @@ export class SidebarComponent {
...
+          localMapMeta.objectUrl = undefined;
+          localMapMeta.mapStatus = 'downloaded';
+          this.mapDownloadStates[map] = localMapMeta.mapStatus;
+          await db.localMapMeta.put(localMapMeta);
+        },
+        error: async (error) => {
+          localMapMeta.mapStatus = 'missing';
+          this.mapDownloadStates[map] = localMapMeta.mapStatus;
+          await db.localMapMeta.put(localMapMeta);
+          console.error('Download error:', error);
```
Comment (jannis-z) [inline src/app/sidebar/sidebar/sidebar.component.ts:153]: remove log

File: src/app/sidebar/sidebar/sidebar.component.ts [hunk 4]
```diff
@@ -114,38 +116,62 @@ export class SidebarComponent {
...
+
+      // Store the subscription so it can be cancelled later
+      this.subscriptions.set(map, mapRequest);
+    }
+  }
+
+  async cancelDownloadMap(map: ZsMapStateSource) {
+    if (this.subscriptions.has(map)) {
+      const subscription = this.subscriptions.get(map);
+      subscription?.unsubscribe();
```
Comment (jannis-z) [inline src/app/sidebar/sidebar/sidebar.component.ts:165]: 164 + 165 can be simplified to 1 line

File: src/app/sidebar/sidebar/sidebar.component.ts [hunk 5]
```diff
@@ -114,38 +116,62 @@ export class SidebarComponent {
...
+      this.subscriptions.delete(map);
+      this.mapDownloadStates[map] = 'missing';
```
Comment (jannis-z) [inline src/app/sidebar/sidebar/sidebar.component.ts:167]: lines 167-173 are not necessary. status is always missing

**Review by jannis-z — CHANGES_REQUESTED** [review]

**Comment by jon4hz** [comment]
superseded by #3

**Pull request closed without merging**